x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/support/WatcherUtilsTests.java

```diff
@@ -90,7 +90,7 @@ public void testResponseToData() throws Exception {
 
     public void testSerializeSearchRequest() throws Exception {
         String[] expectedIndices = generateRandomStringArray(5, 5, true);
-        String[] expectedTypes = generateRandomStringArray(2, 5, true);
+        String[] expectedTypes = generateRandomStringArray(2, 5, true, false);
```

Comment:
This is a very minor point, but I wonder if it would be possible to do a small refactor in `WatcherSearchTemplateRequest#fromXContent` so that an empty types array is not treated as `type == null`?